pkg/kubelet/allocation/allocation_manager.go

```diff
@@ -214,6 +232,37 @@ func (m *manager) Run(ctx context.Context) {
 	}()
 }
 
+// Gernerate pod resize completed event
+func (m *manager) emitPodResizeCompletionEvent(resizedPod *v1.Pod) string {
+	podResizeSource := &podResourceSummary{
+		InitContainers: make([]containerAllocation, 0),
+		Containers:     make([]containerAllocation, 0),
+	}
+
+	podutil.VisitContainers(&resizedPod.Spec, podutil.InitContainers|podutil.Containers,
+		func(allocatedContainer *v1.Container, containerType podutil.ContainerType) bool {
+			allocatedResources, exists := m.GetContainerResourceAllocation(resizedPod.UID, allocatedContainer.Name)
+			if exists {
+				allocation := containerAllocation{
+					Name:      allocatedContainer.Name,
+					Resources: allocatedResources,
+				}
+
+				switch containerType {
+				case podutil.InitContainers:
+					podResizeSource.InitContainers = append(podResizeSource.InitContainers, allocation)
+				case podutil.Containers:
+					podResizeSource.Containers = append(podResizeSource.Containers, allocation)
+				}
+			}
+			return true
+		})
+
+	podResizeMsgDetailsJSON, _ := json.Marshal(podResizeSource)
```

Comment:
Don't ignore the error here. Maybe something like this:


```suggestion
	podResizeMsgDetailsJSON, err := json.Marshal(podResizeSource)
	if err != nil {
		klog.ErrorS(err, "Failed to serialize resource summary", "pod", format.Pod(pod))
		return "Pod resize completed"
	}
```